cdc/redo/reader/reader.go

```diff
@@ -22,30 +22,30 @@ import (
 	"path/filepath"
 	"sync"
 
-	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/cdc/redo/common"
-	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"github.com/pingcap/tiflow/pkg/errors"
 	"github.com/pingcap/tiflow/pkg/redo"
+	"github.com/pingcap/tiflow/pkg/sink/mysql"
 	"go.uber.org/multierr"
 	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	emitBatch             = mysql.DefaultMaxTxnRow
+	defaultReaderChanSize = mysql.DefaultWorkerCount * emitBatch
 )
 
 // RedoLogReader is a reader abstraction for redo log storage layer
 type RedoLogReader interface {
-	io.Closer
-
-	// ResetReader setup the reader boundary
-	ResetReader(ctx context.Context, startTs, endTs uint64) error
-
-	// ReadNextLog reads up to `maxNumberOfMessages` messages from current cursor.
-	// The returned redo logs sorted by commit-ts
-	ReadNextLog(ctx context.Context, maxNumberOfEvents uint64) ([]*model.RowChangedEvent, error)
-
-	// ReadNextDDL reads `maxNumberOfDDLs` ddl events from redo logs from current cursor
-	ReadNextDDL(ctx context.Context, maxNumberOfEvents uint64) ([]*model.DDLEvent, error)
-
+	// Run read and decode redo logs in background.
+	Run(ctx context.Context) error
+	// ReadNextRow read one row event from redo logs.
+	ReadNextRow(ctx context.Context) (*model.RowChangedEvent, error)
```

Comment:
Before this pr, what is the batch size that `ReadNextLog` return in most cases? Is there any performance issue that changing it from batch to one row?

Reply:
Currently we use [buffered channel](https://github.com/pingcap/tiflow/pull/8362/files#diff-7194f7523941d867fef7ed626c77db0337996b8bc35063b9fa595099215f51beR108) to cache logs with [the same capacity](https://github.com/pingcap/tiflow/pull/8362/files#diff-7194f7523941d867fef7ed626c77db0337996b8bc35063b9fa595099215f51beR38) as the previous batchsize.